meson/post_install.py

```diff
@@ -1,29 +1,35 @@
 #!/usr/bin/env python3
 
-import os
+from os import path, environ
 import subprocess
 import sys
 
 project_name = sys.argv[1]
-prefix = os.environ['MESON_INSTALL_PREFIX']
-schemadir = os.path.join(prefix, 'share', 'glib-2.0', 'schemas')
-schema_file = os.path.join(schemadir, project_name + '.gschema.xml')
-mimedir = os.path.join(prefix, 'share', 'mime')
+prefix = environ.get('MESON_INSTALL_PREFIX', '/usr/local')
+datadir = path.join(prefix, 'share')
+schemadir = path.join(datadir, 'glib-2.0', 'schemas')
+schema_file = path.join(schemadir, project_name + '.gschema.xml')
+desktop_database_dir = path.join(datadir, 'applications')
+mimedir = path.join(datadir, 'mime')
 
 def fill_schema_placeholders():
     with open(schema_file, 'rt') as file:
         schema_content = file.read()
-    for param in os.environ.keys():
+    for param in environ.keys():
         key = '{{' + param + '}}'
-        value = os.environ[param]
+        value = environ[param]
         schema_content = schema_content.replace(key, value)
     with open(schema_file, 'wt') as file:
         file.write(schema_content)
 
-if not os.environ.get('DESTDIR'):
-    print('Replacing gsettings placeholders in ' + schema_file + '...')
+if not environ.get('DESTDIR'):
+    print('Replacing gsettings placeholders in ' + schema_file + '…')
     fill_schema_placeholders()
-    print('Compiling gsettings schemas...')
+    print('Compiling gsettings schemas…')
     subprocess.call(['glib-compile-schemas', schemadir])
-    print('Registering text/gemini MIME type...')
+    print('Updating desktop database…')
+    subprocess.call(['update-desktop-database', '-q', desktop_database_dir])
+    print('Updating icon cache…')
+    subprocess.call(['gtk-update-icon-cache', '-qtf', path.join(datadir, 'icons', 'hicolor')])
```

Comment:
Are these manual updates necessary? I'm reluctant to add install steps which would make installation take longer for end-users.

Reply:
It's the same concept as the `update-mime-database` and `glib-compile-schemas` steps you are already running there.

They are necessary in order to make desktop files or icons actually get picked up on the system, because the system has a fast lookup cache that needs to be updated every time installed files are added/changed/removed.

Granted, you could use a somewhat optimized version of these commands here: https://mesonbuild.com/Gnome-module.html#gnomepost_install
Requires Meson 0.57 though.

Reply:
Ok, tnx for the clarification.